src/server/raftkv.rs

```diff
@@ -201,10 +200,12 @@ where
         cb: Callback<CmdRes<E::Snapshot>>,
     ) -> Result<()> {
         let mut header = self.new_request_header(ctx.pb_ctx);
-        if ctx.pb_ctx.get_stale_read() && !ctx.start_ts.is_zero() {
+        if ctx.pb_ctx.get_stale_read()
+            && (ctx.start_ts.is_none() || !ctx.start_ts.as_ref().unwrap().is_zero())
```

Comment:
You mean `ctx.start_ts.map_or(true, |ts| !ts.is_zero())`?

Reply:
yes